api/engineapi/contracts_actions.py

```diff
@@ -0,0 +1,633 @@
+import argparse
+import json
+import logging
+import uuid
+from enum import Enum
+from typing import Any, Dict, List, Optional, Tuple
+
+from sqlalchemy import func, text
+from sqlalchemy.exc import IntegrityError, NoResultFound
+from sqlalchemy.orm import Session
+from web3 import Web3
+
+from . import data, db
+from .models import RegisteredContract, CallRequest
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class ContractAlreadyRegistered(Exception):
+    pass
+
+
+class ContractType(Enum):
+    raw = "raw"
+    dropper = "dropper-v0.2.0"
+
+
+def validate_method_and_params(
+    contract_type: ContractType, method: str, parameters: Dict[str, Any]
+) -> None:
+    """
+    Validate the given method and parameters for the specified contract_type.
+    """
+    if contract_type == ContractType.raw:
```

Comment:
I think it should have `else` with raise Exception if incorrect contract type specified, otherwise UniqueConstraint will not affect.